v0/destinations/salesforce_pardot/transform.js

```diff
@@ -0,0 +1,284 @@
+const axios = require("axios");
+const {EventType} = require("../../../constants");
+const {
+    SF_TOKEN_REQUEST_URL,
+    SFPARDOT_API_REQUEST_URL,
+    ignoredTraits, SFPARDOT_API_VERSION
+} = require("./config");
+const {
+    defaultRequestConfig,
+    defaultPostRequestConfig,
+    getFieldValueFromMessage,
+    getSuccessRespEvents,
+    getErrorRespEvents,
+    CustomError
+} = require("../../util");
+const logger = require("../../../logger");
+
+// Utility method to construct the header to be used for SFDC API calls
+// The "Authorization: Bearer <token>" header element needs to be passed for
+// authentication for all SFDC REST API calls
+async function getSFPardotHeader(destination) {
+
+    const authUrl = `${SF_TOKEN_REQUEST_URL}?username=${
+        destination.Config.userName
+    }&password=${encodeURIComponent(
+        destination.Config.password
+    )}&client_id=${
+        destination.Config.consumerKey
+    }&client_secret=${destination.Config.consumerSecret}&grant_type=password`;
+
+    let response;
+    try {
+        response = await axios.post(authUrl, {});
+    } catch (error) {
+        throw new CustomError(
+            `SALESFORCE AUTH FAILED: ${error.message}`,
+            error.status || 400
+        );
+    }
+
+    return {
+        token: `Bearer ${response.data.access_token}`,
+        instanceUrl: response.data.instance_url,
+        businessUnitId: destination.Config.businessUnitId
+    };
+}
+
+function responseBuilderSimple(
+    traits,
+    prospectMap,
+    authorizationData
+) {
+
+    let pardotCreateEndPointURL = `${SFPARDOT_API_REQUEST_URL}/prospect/version/4/do/create?format=json`;
+    let pardotUpdateEndPointURL = `${SFPARDOT_API_REQUEST_URL}/prospect/version/4/do/update/id/`;
+    const response = defaultRequestConfig();
+
+    const header = {
+        Authorization: authorizationData.token,
+        "Pardot-Business-Unit-Id": authorizationData.businessUnitId,
+        "Content-Type": "application/x-www-form-urlencoded"
+    };
+
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.headers = header;
+
+    if( prospectMap) {
+        // Update existing Prospect REQUEST
+        logger.info('Update Prospect request for: ' + traits.email + ' with TenantId: ' + traits.tenantId);
+        pardotEndPointURL = pardotUpdateEndPointURL + prospectMap.prospectId + `?format=json`;
+
+    } else {
+        // Create New Prospect
+        logger.info('Create Prospect request for: ' + traits.email + ' with TenantId: ' + traits.tenantId);
+        pardotEndPointURL = pardotCreateEndPointURL;
+    }
+
+    response.body.FORM = {
+        email: traits.email,
+        first_name: traits.first_name,
+        last_name: traits.last_name,
+        tenantID: traits.tenantId,
+        phone: traits.phone
```

Comment:
have you tested the case where phone is blank

Reply:
yes